### ISSUE TITLE
Recent calendar app versions break buttons in embedded calendars

### ISSUE THREAD

**Issue by nursoda** [nextcloud/calendar#5243]
### Steps to reproduce

1. create a calendar intended to be shared to the public
2. actually share it and copy the "embedding code"
3. use that code in some website such as https://wir.freiburg.social/termine/ or https://suevia-ka.de/veranstaltungen/
4. try to switch the view (Day/Week/Month/List) or select next/previous

### Expected behavior

Buttons should work (they did until recently!)

### Actual behaviour

NC Name is written over the "current week/month", buttons don't work:
![grafik](https://github.com/nextcloud/calendar/assets/8425483/4353dd5e-b180-4dad-b739-30e70074db76)

### Calendar app version

4.3.4

### CalDAV-clients used

Just the NC Calendar App Web View

### Browser

Firefox 113.0.1

### Client operating system

Arch Linux

### Server operating system

Arch Linux

### Web server

Nginx

### Database engine version

MariaDB

### PHP engine version

PHP 8.2

### Nextcloud version

26.0.1

### Updated from an older installed version or fresh install

Updated from an older version

### List of activated apps

```Text
Enabled:
  - activity: 2.18.0
  - bookmarks: 13.0.1
  - bruteforcesettings: 2.6.0
  - calendar: 4.3.4
  - camerarawpreviews: 0.8.1
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - contacts: 5.2.0
  - dav: 1.25.0
  - deck: 1.9.2
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - groupfolders: 14.0.2
  - impersonate: 1.13.0
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - notes: 4.7.2
  - notifications: 2.14.0
  - notify_push: 0.6.3
  - oauth2: 1.14.0
  - password_policy: 1.16.0
  - photos: 2.2.0
  - polls: 5.0.5
  - previewgenerator: 5.3.0
  - privacy: 1.10.0
  - provisioning_api: 1.16.0
  - related_resources: 1.1.0-alpha1
  - richdocuments: 8.0.2
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - spreed: 16.0.3
  - suspicious_login: 4.4.0
  - systemtags: 1.16.0
  - tasks: 0.15.0
  - text: 3.7.2
  - theming: 2.1.1
  - theming_customcss: 1.14.0
  - twofactor_backupcodes: 1.15.0
  - twofactor_email: 2.7.2
  - twofactor_nextcloud_notification: 3.7.0
  - twofactor_totp: 8.0.0-alpha.0
  - twofactor_webauthn: 1.2.0
  - updatenotification: 1.16.0
  - viewer: 1.10.0
  - workflowengine: 2.8.0
Disabled:
  - admin_audit: 1.16.0
  - comments: 1.16.0 (installed 1.14.0)
  - contactsinteraction: 1.7.0 (installed 1.1.0)
  - dashboard: 7.6.0 (installed 7.0.0)
  - encryption: 2.14.0
  - files_external: 1.18.0
  - files_mindmap: 0.0.27 (installed 0.0.27)
  - firstrunwizard: 2.15.0 (installed 2.6.0)
  - mail: 3.1.1 (installed 3.1.1)
  - nextcloud_announcements: 1.15.0 (installed 1.6.0)
  - ransomware_protection: 1.14.0 (installed 1.14.0)
  - recommendations: 1.5.0 (installed 0.5.0)
  - sharerenamer: 3.2.0 (installed 3.2.0)
  - support: 1.9.0 (installed 1.0.1)
  - survey_client: 1.14.0 (installed 1.5.0)
  - twofactor_gateway: 0.20.0 (installed 0.20.0)
  - user_ldap: 1.16.0
  - user_status: 1.6.0 (installed 1.0.0)
  - weather_status: 1.6.0 (installed 1.0.0)
```

### Nextcloud configuration

```Text
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.suevia-ka.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "26.0.1.1",
        "overwrite.cli.url": "https:\/\/cloud.suevia-ka.de",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "dbindex": 4,
            "timeout": 1.5
        },
        "logtimezone": "Europe\/Berlin",
        "trashbin_retention_obligation": "auto,30",
        "versions_retention_obligation": "auto,366",
        "skeletondirectory": "",
        "default_language": "de",
        "default_locale": "de_DE",
        "default_phone_region": "DE",
        "updater.release.channel": "stable",
        "theme": "",
        "loglevel": 2,
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "mail_smtpport": "25",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "simpleSignUpLink.shown": false,
        "twofactor_enforced": "true",
        "twofactor_enforced_groups": [
            "Gruppenadmin",
            "admin"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false
    }
}
```

### Web server error log

```Text
not relevant
```

### Log file

```Text
only entries not relevant to calendar (group folder deprecation warning)
```

### Browser log

```Text
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf inline blockiert ("script-src"). single-file-extension-frames.js:1:294
Object { source: "iframe.js", func: "onAd", args: (1) […] }
driver.js:119:18
Object { source: "content.js", func: "isDisabledDomain", args: (1) […] }
driver.js:119:18
Object { hostname: "cloud.suevia-ka.de", technologies: (25) […] }
driver.js:119:18
JQMIGRATE: Migrate is installed, version 3.4.0 core-common.js:2:10097392
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 core-main.js:2:66595
Diese Seite verwendet die nicht standardisierte Eigenschaft "zoom". Stattdessen sollte calc() in den entsprechenden Eigenschaftswerten oder "transform" zusammen mit "transform-origin: 0 0" verwendet werden. 2023-04-01
You need to fill either the text or the ariaLabel props in the button component. 
Object { text: undefined, ariaLabel: null }
 
Object { _uid: 16, _isVue: true, __v_skip: true, _scope: {…}, "$options": {…}, _renderProxy: {…}, _self: {…}, "$parent": {…}, "$root": {…}, "$children": [], … }
calendar-main.js:2:2265790
@nextcloud/calendar-js app is using version 2.2019c of the timezone database calendar-main.js:2:193888
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 3 core-main.js:2:66595
session heartbeat polling started core-main.js:2:50881
[ERROR] Files_PDFViewer: But this does not appear to be a public page 
Object { app: "Files_PDFViewer", level: 2 }
files_pdfviewer-public.js:2:11771
You need to fill either the text or the ariaLabel props in the button component. 
Object { text: undefined, ariaLabel: null }
 
Object { _uid: 16, _isVue: true, __v_skip: true, _scope: {…}, "$options": {…}, _renderProxy: {…}, _self: {…}, "$parent": {…}, "$root": {…}, "$children": (1) […], … }
calendar-main.js:2:2265790
Object { source: "content.js", func: "onContentLoad", args: (3) […] }
driver.js:119:18
Object { source: "driver.js", func: "analyzeRequires", args: (2) […] }
driver.js:119:18
Object { source: "content.js", func: "onContentLoad", args: (5) […] }
driver.js:119:18
Object { source: "content.js", func: "onContentLoad", args: (5) […] }
driver.js:119:18
Object { source: "content.js", func: "onContentLoad", args: (5) […] }
driver.js:119:18
Object { source: "content.js", func: "onContentLoad", args: (5) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeJs", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeDom", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeJs", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeDom", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeJs", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeDom", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeJs", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeDom", args: (4) […] }
driver.js:119:18
Object { hostname: "cloud.suevia-ka.de", technologies: (25) […] }
driver.js:119:18
Object { source: "content.js", func: "getTechnologies", args: [] }
driver.js:119:18
Backbone is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 6 core-main.js:2:66595
moment is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
oc_config is deprecated: use OC.config instead, this will be removed in Nextcloud 20 core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
Object { source: "content.js", func: "analyzeJs", args: (4) […] }
driver.js:119:18
Object { source: "content.js", func: "analyzeDom", args: (4) […] }
driver.js:119:18
Object { source: "driver.js", func: "analyzeRequires", args: (2) […] }
driver.js:119:18
Object { source: "driver.js", func: "analyzeRequires", args: (2) […] }
driver.js:119:18
Object { hostname: "cloud.suevia-ka.de", technologies: (25) […] }
driver.js:119:18
Object { hostname: "cloud.suevia-ka.de", technologies: (25) […] }
driver.js:119:18
Backbone is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 6 core-main.js:2:66595
moment is deprecated: please ship your own, this will be removed in Nextcloud 20 2 core-main.js:2:66595
oc_config is deprecated: use OC.config instead, this will be removed in Nextcloud 20 core-main.js:2:66595
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. core-main.js:2:66595
Object { source: "content.js", func: "analyzeJs", args: (2) […] }
driver.js:119:18
Object { source: "driver.js", func: "analyzeRequires", args: (2) […] }
driver.js:119:18
Object { hostname: "cloud.suevia-ka.de", technologies: (25) […] }
driver.js:119:18
```

### Additional info

Using this CustomCSS:

:root { --body-container-margin: 0px; --body-container-radius: 0px; }
* { --border-radius-pill:10px; }

However, I verified that deleting all CustomCSS also yields the issue.

**Comment by alexanderdd** [comment]
I can confirm:

![Bildschirmfoto vom 2023-05-23 21-16-26](https://github.com/nextcloud/calendar/assets/28591861/3d82d483-f42d-4edc-aef9-9d3be1ce35b9)

The user-selected logo and user-entered name of the Nextcloud is shown on top of the buttons. Not usable anymore.

Nextcloud 25.0.6
calendar 4.3.4
Firefox 113.0.1 (but also in latest Chromium)

**Comment by alexanderdd** [comment]
Maybe it's related to this recent change? https://github.com/nextcloud/calendar/pull/5184

@tcitworld  maybe you can look at this?

**Comment by tcitworld** [comment]
See https://github.com/nextcloud/calendar/pull/5251

**Comment by Luncheon3462** [comment]
@nursoda I'm on nc 27.1.2 with calendar 4.5.2 and when trying to embed a calendar onto a different domain I control, I get a nextcloud page that says “page not found on the server or you may not be allowed to view it”

How did you even get the calendar to show up?    
<img width="601" alt="Screenshot 2023-11-04 at 10 39 57 AM" src="https://github.com/nextcloud/calendar/assets/140646544/c39d08c3-bd19-4b02-8859-7e184f70339e">

**Comment by nursoda** [comment]
This is off-topic for this issue, but…

Steps to share a calendar are described in the [user manual](https://docs.nextcloud.com/server/latest/user_manual/en/groupware/calendar.html#publishing-a-calendar). As the messages above seems to be issued by your NC, it's not an issue of deep-link prohibition, so I assume you just have not created a public link from the three-dot-menu of the calendar to share within the left pane in the calendar app?

Please continue this in the [forum](https://help.nextcloud.com/) and be a bit more verbose: Describe which documentation you followed, which steps you took, what you expected and what you got instead.

**Comment by nursoda** [comment]
The issue seems to be resolved:
![grafik](https://github.com/nextcloud/calendar/assets/8425483/36755bc9-8f12-4452-be48-ecd00d99fbff)
This is on NC 27.1.3

**Comment by nursoda** [comment]
@Luncheon3462 In case you didn't see my EDITED post, here's a heads up. I also found a (workaround?) solution that might help you: Use [CSP editor](https://apps.nextcloud.com/apps/csp_editor) to allow the site you want to embed your link in. However, at least for me, there's an [open issue](https://framagit.org/framasoft/nextcloud/csp_editor/-/issues/7) using it (and a workaround for that, too). To be honest, I expect a **public** link to **not** be protected from embedding, so that's probably a bug. Would you report that bug, please, and CC me?

**Comment by alexanderdd** [comment]
I can confirm the original issue is fixed in NC 26.0.6
